static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -454,6 +459,21 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
     changedTriggerIndex?: number
   ) {
     const {comparisonType} = this.state;
+    // If we have an anomaly detection alert, then we don't need to validate the thresholds, but we do need to set them to 0
+    if (comparisonType === AlertRuleComparisonType.DYNAMIC) {
+      // NOTE: we don't support warning triggers for anomaly detection alerts yet
+      // once we do, uncomment this code and delete the next five uncommented lines:
+      // triggers.forEach(trigger => {
+      //   trigger.alertThreshold = 0;
+      // });
+      const criticalTriggerIndex = triggers.findIndex(
+        ({label}) => label === AlertRuleTriggerType.CRITICAL
+      );
+      const criticalTrigger = triggers[criticalTriggerIndex];
+      criticalTrigger.alertThreshold = 0;
```

Comment:
Can you explain this bit of code a little? (It looks like we're taking the `triggers` from the state, then mutating the variable to trigger state changes in the component)

Reply:
Dynamic alerts don't have trigger thresholds, so we don't have any fields where the user can modify the threshold. On submitting a dynamic alert, the `alertThreshold` will be empty, its default state (because the user never entered a threshold value). Because the trigger needs to have an alert threshold, we set it to 0 here.